swabbie-web/src/main/kotlin/com/netflix/spinnaker/swabbie/controllers/ResourceController.kt

```diff
@@ -121,73 +140,99 @@ class ResourceController(
   }
 
   /**
-   * Runs resource through marking logic, calculating if it is a candidate for deletion.
-   * Note: this api is quite slow, because does the actual recalculation for a resource.
+   * FOR TESTING
+   * Marks an image for deletion with the given soft delete and deletion times.
+   * Body contains information needed for marking
    */
-  @RequestMapping(value = ["/check/{namespace}/{resourceId}"], method = [RequestMethod.GET])
-  fun checkResource(
+  @RequestMapping(value = ["/state/{namespace}/{resourceId}/mark"], method = [RequestMethod.PUT])
+  fun onDemandMark(
+    @PathVariable resourceId: String,
     @PathVariable namespace: String,
-    @PathVariable resourceId: String
-  ): ResourceEvauation {
-    val workConfiguration = getWorkConfiguration(namespaceParser(namespace))
+    @RequestBody markInformation: OnDemandMarkData
+  ) {
+    val workConfiguration = findWorkConfiguration(SwabbieNamespace.namespaceParser(namespace))
+    val handler = resourceTypeHandlers.find { handler ->
+      handler.handles(workConfiguration)
+    } ?: throw NotFoundException("No handlers for $namespace")
+
+    handler.markResource(resourceId, markInformation, workConfiguration)
+  }
 
+  /**
+   * FOR TESTING
+   * Soft deletes a resource
+   */
+  @RequestMapping(value = ["/state/{namespace}/{resourceId}/softDelete"], method = [RequestMethod.PUT])
+  fun softDelete(
+    @PathVariable resourceId: String,
+    @PathVariable namespace: String
+  ) {
+    val workConfiguration = findWorkConfiguration(SwabbieNamespace.namespaceParser(namespace))
     val handler = resourceTypeHandlers.find { handler ->
       handler.handles(workConfiguration)
     } ?: throw NotFoundException("No handlers for $namespace")
 
-    return handler.evaluateCandidate(resourceId, resourceId, workConfiguration)
+    handler.softDeleteResource(resourceId, workConfiguration)
   }
 
-  internal fun getWorkConfiguration(
-    namespace: Namespace
-  ): WorkConfiguration {
-    return WorkConfigurator(
-      swabbieProperties = generateSwabbieProperties(namespace),
-      accountProvider = accountProvider,
-      exclusionPolicies = listOf(AccountExclusionPolicy()),
-      exclusionsSuppliers = Optional.empty()
-    ).generateWorkConfigurations()[0]
+  /**
+   * FOR TESTING
+   * Deletes a resource
+   */
+  @RequestMapping(value = ["/state/{namespace}/{resourceId}/delete"], method = [RequestMethod.PUT])
+  fun delete(
+    @PathVariable resourceId: String,
+    @PathVariable namespace: String
+  ) {
+    val workConfiguration = findWorkConfiguration(SwabbieNamespace.namespaceParser(namespace))
+    val handler = resourceTypeHandlers.find { handler ->
+      handler.handles(workConfiguration)
+    } ?: throw NotFoundException("No handlers for $namespace")
+
+    handler.deleteResource(resourceId, workConfiguration)
```

Comment:
This delete & softdelete methods don't feel right.

Why not have a single endpoint:

```
@RequestMapping(value = ["/state/{namespace}/{resourceId}"], method = [RequestMethod.DELETE])
fun delete(
  @PathVariable resourceId: String,
  @PathVariable namespace: String,
  @QueryParam(required = true) soft: Boolean
)
```

Reply:
Good idea will change